src/sentry/api/bases/organization.py

```diff
@@ -91,6 +103,119 @@ class OrganizationDiscoverSavedQueryPermission(OrganizationPermission):
 
 class OrganizationEndpoint(Endpoint):
     permission_classes = (OrganizationPermission, )
+    # Defines what happens if no date filter parameters are passed. If False,
+    # no date filtering occurs. If True, we provide default values.
+    DATE_FILTER_OPTIONAL = False
+
+    def get_project_ids(self, request, organization, force_global_perms=False):
+        """
+        Determines which project ids to filter the endpoint by. If a list of
+        project ids is passed in via the `project` querystring argument then
+        validate that these projects can be accessed. If not passed, then
+        return all project ids that the user can access within this
+        organization.
+
+        :param request:
+        :param organization: Organization to fetch projects for
+        :param force_global_perms: Permission override. Allows subclasses to
+        perform their own validation and allow the user to access any project
+        in the organization. This is a hack to support the old
+        `request.auth.has_scope` way of checking permissions, don't use it
+        for anything else, we plan to remove this once we remove uses of
+        `auth.has_scope`.
+        :return: A list of project ids, or raises PermissionDenied.
+        """
+        project_ids = set(map(int, request.GET.getlist('project')))
+
+        requested_projects = project_ids.copy()
+
+        om_role = None
+        if request.user.is_authenticated():
+            try:
+                om_role = OrganizationMember.objects.filter(
+                    user=request.user,
+                    organization=organization,
+                ).values_list('role', flat=True).get()
+            except OrganizationMember.DoesNotExist:
+                pass
+
+        if (
+            request.user.is_superuser
+            or (om_role and roles.get(om_role).is_global)
+            or organization.flags.allow_joinleave
```

Comment:
hmmm wondering if this should actually be here. i don't know that we've been super consistent about how this works in the product right now -- basically should a user see all projects they _could_ access or just the projects they currently have access to. maybe for now this should be another thing the endpoints can set themselves and then we can work on getting a decision on how the product should behave and we can figure out how to adjust accordingly.

Reply:
Yeah, I'm not sure what is correct here. You're probably right that it should be parameterized, will add that in.